python/az/aro/azext_aro/custom.py

```diff
@@ -455,6 +464,20 @@ def get_network_resources_from_subnets(cli_ctx, subnets, fail):
         if subnet.get("natGateway", None):
             subnet_resources.add(subnet['natGateway']['id'])
 
+        if oc.network_profile.preconfigured_nsg == 'Enabled':
+            if subnet.get("networkSecurityGroup", None):
+                subnet_resources.add(subnet['networkSecurityGroup']['id'])
+            else:
+                subnets_with_no_nsg_attached.add(sn)
+
+    # when preconfiguredNSG feature is Enabled we either have all subnets NSG attached or none.
+    if oc.network_profile.preconfigured_nsg == 'Enabled' and \
+        len(subnets_with_no_nsg_attached) != 0 and \
+            len(subnets_with_no_nsg_attached) != len(subnets):
+        raise ValidationError(f"(ValidationError) preconfiguredNSG feature is enabled but NSG not attached for subnets.\
+                                  Please make sure all the following subnets have network security group attached and retry.\
```

Comment:
one more grammar issue, should be either:

1. `have a network` not `have network` (if only one NSG is to be used), or
2. `groups` not `group` (if multiple NSGs can be used)